Scenarios/AKS-Secure-Baseline-PrivateCluster/Terraform/11-blue-green.md

```diff
@@ -0,0 +1,268 @@
+# Blue Green Deployment
+
+Here you find the details of the steps to adopt the blue green deployment pattern. The reference implementation is based on the existing secure baseline implementation with the addition of new data structure to add abstraction to manage multiple AKS clusters, in this case 2 AKS clusters, the blue and green, and additional supporting azure resources like: Application Gateway and Public DNS Zone.
+
+The new introduced data structures are:
+
+- Application Gateway map to deploy the blue and green Application Gateways, there is a one to one mapping between the AKS clusters and Application Gateways due to the adoption of the AGIC addon for AKS.
+Below the data structure and related documentation. By default only the blue application gateway is deployed.
+
+```
+locals {
+  Map of the aure application gateway to deploy
+  appgws = {
+    "appgw_blue" = {
+      prefix used to configure uniques names and parameter values
+      name_prefix="blue"
+      Boolean flag that enable or disable the deployment of the specific application gateway
+      appgw_turn_on=true
+    },
+    "appgw_green" = {
+      name_prefix="green"
+      appgw_turn_on=false
+    }
+  }
+}
+```
+
+- The other map structure is dedicated to the AKS clusters. Below the data structure and related documentation. By default only the blue AKS cluster is deployed.
+
+```
+locals {
+  Map of the AKS Clusters to deploy
+  aks_clusters = {
+    "aks_blue" = {
+      prefix used to configure unique names and parameter values
+      name_prefix="blue"
+      Boolean flag that enable or disable the deployment of the specific AKS cluster
+      aks_turn_on=true
+      The kubernetes version to use on the cluster
+      k8s_version="1.23.5"
+      Reference Name to the Application gateway that need to be associaated to the AKS Cluster with the AGIC addo-on
+      appgw_name="lzappgw-blue"
+    },
+    "aks_green" = {
+      name_prefix="green"
+      aks_turn_on=false
+      k8s_version="1.23.8"
+      appgw_name="lzappgw-green"
+    }
+  }
+}
+```
+
+As part of the blue green deployment there is also a dedicated public DNS zone to register the 3 hostnames required to implment the pattern:
+- Public facing hostname, the one used by the end users of the workloads/apps hosted into the clusters
+- blue cluster hostname, that is dedicated fot the internal validation
+- green cluster hostname, that is dedicated fot the internal validation
+
```

Comment:
Would this be a good place to tell then to run the terraform module located here "08-DNS-Records"?

Reply:
This part of the documentation is more abstract to describe the principal aspects of the pattern, the reference to the DNS records registration is at "## T3: Traffic Switch to the green cluster". Anyway I have re-worded to be more clear "As part of the blue green deployment there is the condifguration of 3 hostnames required to implement the pattern:
- Public facing hostname, the one used by the end users of the workloads/apps hosted into the clusters
- blue cluster hostname, that is dedicated for the internal validation
- green cluster hostname, that is dedicated for the internal validation

As part of the solution there is also the optional deployment of a public DNS zone to manage in azure the DNS records for the custom domain assigned to  the applications.
you can configure the DNS records in the public DNS zone using the following [guide](./09-dns-records.md)
"